.github/workflows/build.yaml

```diff
@@ -31,21 +31,21 @@ jobs:
       packages: write
     steps:
       - name: Checkout (GitHub)
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: Login to GitHub Container Registry
-        uses: docker/login-action@v2
+        uses: docker/login-action@v3
         with:
           registry: ghcr.io
           username: ${{ github.repository_owner }}
-          password: ${{ secrets.GITHUB_TOKEN }}
+          password: ${{ github.token }}
 
       - name: Build and run Dev Container task
         uses: devcontainers/ci@v0.3
         with:
           imageName: ghcr.io/mt-caret/polars-ocaml
           cacheFrom: ghcr.io/mt-caret/polars-ocaml
-          push: always
+          refFilterForPush: refs/heads/main
```

Comment:
This should prevent CI from failing: https://github.com/devcontainers/ci/blob/8b2e2d0300e69c8ca05b66f27195d4559344cb3a/action.yml#L36-L39